src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -1637,17 +1665,38 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
     // Now if all of the tasks are blocked, then we need to break a deadlock
     bool ret = all_task_ids.size() == blocked_task_ids.size() && !all_task_ids.empty();
     if (ret) {
+      std::set<long> threadsKeySet;
+      std::transform(threads.begin(),
+                     threads.end(),
+                     std::inserter(threadsKeySet, threadsKeySet.begin()),
+                     [](const auto& pair) { return pair.first; });
+
       logger->info(
-        "deadlock state is reached with all_task_ids size: {}, blocked_task_ids: {}, "
-        "bufn_task_ids: {}, threads size: {}",
+        "deadlock state is reached with all_task_ids: {} ({}), blocked_task_ids: {} ({}), "
+        "bufn_task_ids: {} ({}), threads: {} ({})",
+        setToString(all_task_ids),
```

Comment:
seems like this is only used when there is a deadlock: hope it's not often. The reason I mention this is that even if the logger level is set to warn/error we will still call `setToString` and build a string that won't be printed anywhere.

Reply:
hi @abellina ,  I will protect them with `if(is_log_enabled)`

Reply:
fixed